lib/rules/prefer-aria-over-title-attribute.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Rule } from "eslint";
+import { JSXAttribute, JSXOpeningElement, Literal } from "estree-jsx";
+import { elementType, hasProp } from "jsx-ast-utils";
+import { hasAssociatedLabelViaAriaLabelledBy } from "../util/labelUtils";
+import { hasNonEmptyProp } from "../util/hasNonEmptyProp";
+import { hasToolTipParent } from "../util/hasTooltipParent";
+import { hasTextContentChild } from "../util/hasTextContentChild";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const applicableComponents = ["Button"];
+
+const rule: Rule.RuleModule = {
+    meta: {
+        // possible error messages for the rule
+        messages: {
+            preferAria: `Prefer aria over the title attribute for accessible labelling: ${applicableComponents.join(", ")}`
+        },
+        type: "suggestion", // `problem`, `suggestion`, or `layout`
+        docs: {
+            description:
+                "The title attribute is not consistently read by screen readers, and its behavior can vary depending on the screen reader and the user's settings.",
+            recommended: true
+        },
+        fixable: "code", // Or `code` or `whitespace`
+        schema: [] // Add a schema if the rule has options
+    },
+
+    create(context) {
```

Comment:
I think the context object has a type as well

Reply:
this is inferred by default. all the auto completion benefits will be available on usage